src/Nitric/Faas/Faas.php

```diff
@@ -61,18 +64,32 @@ function () use ($handler, $address) {
                         function (Request $request) use ($handler) {
                             $body = yield $request->getBody()->buffer();
 
+                            $triggerRequest = new TriggerRequest();
+                            // Get the trigger request from the request body
+                            $triggerRequest->mergeFromJsonString($body);
+
                             // Convert HTTP Request to Nitric Request
-                            $nitricRequest = \Nitric\Faas\Request::fromHTTPRequest(
-                                $request->getHeaders(),
-                                $body,
-                                $request->getUri()->getPath()
+                            $nitricRequest = \Nitric\Faas\Request::fromTriggerRequest(
+                                $triggerRequest
                             );
 
-                            // Call the handler function
                             $nitricResponse = $handler($nitricRequest);
+                            if ($nitricResponse instanceof \Nitric\Faas\Response) {
+                                // Return the Nitric Response as an HTTP Response
+                                return self::httpResponse($nitricResponse);
+                            } else {
+                                $stringData = "";
+                                if (is_array($nitricResponse)) {
```

Comment:
Probably also want to check for instance of stdClass and convert those to JSON too